host/class/uac/usb_host_uac/test_app/main/test_host_uac.c

```diff
@@ -0,0 +1,904 @@
+        .stack_size = 4096,
+        .core_id = 0,
+        .callback = uac_host_lib_callback,
+        .callback_arg = NULL
+    };
+
+    ESP_ERROR_CHECK(uac_host_install(&uac_config));
+    ESP_LOGI(TAG, "UAC Class Driver installed");
+}
+
+void test_uac_queue_reset(void)
+{
+    xQueueReset(s_event_queue);
+}
+
+void test_uac_teardown(bool force)
+{
+    if (force) {
+        force_conn_state(false, pdMS_TO_TICKS(1000));
+    }
+    vTaskDelay(500);
+    // uninstall uac host driver
+    ESP_LOGI(TAG, "UAC Driver uninstall");
+    ESP_ERROR_CHECK(uac_host_uninstall());
+    // Wait for USB lib task to finish
+    xEventGroupWaitBits(s_evt_handle, BIT0_USB_HOST_DRIVER_REMOVED, pdTRUE, pdTRUE, portMAX_DELAY);
+    // delete event queue and event group
+    vQueueDelete(s_event_queue);
+    vEventGroupDelete(s_evt_handle);
+    // delay to allow task to delete
+    vTaskDelay(100);
+}
+
+void test_open_mic_device(uint8_t iface_num, uint32_t buffer_size, uint32_t buffer_threshold, uac_host_device_handle_t *uac_device_handle)
+{
+    // check if device params as expected
+    const uac_host_device_config_t dev_config = {
+        .addr = 1,
+        .iface_num = iface_num,
+        .buffer_size = buffer_size,
+        .buffer_threshold = buffer_threshold,
+        .callback = uac_device_callback,
+        .callback_arg = NULL,
+    };
+    ESP_ERROR_CHECK(uac_host_device_open(&dev_config, uac_device_handle));
+}
+
+void test_open_spk_device(uint8_t iface_num, uint32_t buffer_size, uint32_t buffer_threshold, uac_host_device_handle_t *uac_device_handle)
+{
+    // check if device params as expected
+    const uac_host_device_config_t dev_config = {
+        .addr = 1,
+        .iface_num = iface_num,
+        .buffer_size = buffer_size,
+        .buffer_threshold = buffer_threshold,
+        .callback = uac_device_callback,
+        .callback_arg = NULL,
+    };
+    ESP_ERROR_CHECK(uac_host_device_open(&dev_config, uac_device_handle));
+}
+
+void test_close_device(uac_host_device_handle_t uac_device_handle)
+{
+    ESP_ERROR_CHECK(uac_host_device_close(uac_device_handle));
+}
+
+void test_handle_dev_connection(uint8_t *iface_num, uint8_t *if_rx)
+{
+    event_queue_t evt_queue = {0};
+    // ignore the first connected event
+    TEST_ASSERT_EQUAL(pdTRUE, xQueueReceive(s_event_queue, &evt_queue, portMAX_DELAY));
+    TEST_ASSERT_EQUAL(UAC_DRIVER_EVENT, evt_queue.event_group);
+    TEST_ASSERT_EQUAL(1, evt_queue.driver_evt.addr);
+    if (iface_num) {
+        *iface_num = evt_queue.driver_evt.iface_num;
+    }
+    if (if_rx) {
+        *if_rx = evt_queue.driver_evt.event == UAC_HOST_DRIVER_EVENT_RX_CONNECTED ? 1 : 0;
+    }
+}
+
+TEST_CASE("test uac device handling", "[uac_host][known_device]")
+{
+    test_uac_setup();
+    // handle device connection
+    uint8_t mic_iface_num = 0;
+    uint8_t spk_iface_num = 0;
+    test_handle_dev_connection(&mic_iface_num, NULL);
+    TEST_ASSERT_EQUAL(UAC_DEV_MIC_IFACE_NUM, mic_iface_num);
```

Comment:
I got an error here when running this test: `uac_device_handle, volume > 100 ? 100 : volume`
Do I need any special UAC device for testing? 
I am using Hennheiser pc8 headphones

Reply:
@peter-marcisovsky, you are right. the test case label with `[known_device]` aims to check if interfaces be parsed correctlly.

Sorry for the confusion. I will add a comment there.